apps/server/src/modules/board/service/delete-hook.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@nestjs/common';
+import { AnyBoardDo, BoardCompositeVisitorAsync, Card, Column, ColumnBoard, TextElement } from '@shared/domain';
+
+@Injectable()
+export class DeleteHookService implements BoardCompositeVisitorAsync {
+	async visitColumnBoardAsync(columnBoard: ColumnBoard): Promise<void> {
+		await this.visitChildren(columnBoard);
+	}
+
+	async visitColumnAsync(column: Column): Promise<void> {
+		await this.visitChildren(column);
+	}
+
+	async visitCardAsync(card: Card): Promise<void> {
+		await this.visitChildren(card);
+	}
+
+	async visitTextElementAsync(textElement: TextElement): Promise<void> {
+		await this.visitChildren(textElement);
+	}
+
+	async visitChildren(domainObject: AnyBoardDo): Promise<void> {
+		await Promise.all(domainObject.children.map(async (child) => child.acceptAsync(this)));
```

Comment:
I was wondering if we could avoid having a sync and an async accept-method and only having one that is async. And when I saw this line it felt a good idea to remove the sync variant and to rename the acceptAsync to accept.

Reply:
But then you always would have to return a promise even if the `accept` is not async, right? I'm not really sure if that's a good idea. AFAIK it's generally a good practice to distiguish between the two if `async` should be supported.